WordPress/Classes/ViewRelated/NUX/LoginEmailViewController.swift

```diff
@@ -1,35 +1,328 @@
+        emailTextField.text = loginFields.username
+    }
+
+
+    /// Configures whether appearance of the submit button.
+    ///
+    func configureSubmitButton() {
+        submitButton.isEnabled = loginFields.username.isValidEmail()
+    }
+
+
+    /// Sets the view's state to loading or not loading.
+    ///
+    /// - Parameter loading: True if the form should be configured to a "loading" state.
+    ///
+    func configureViewLoading(_ loading: Bool) {
+        emailTextField.isEnabled = !loading
+        submitButton.isEnabled = !loading
+        submitButton.showActivityIndicator(loading)
+    }
+
+
+    /// Configure the view for an editing state. Should only be called from viewWillAppear
+    /// as this method skips animating any change in height.
+    ///
+    func configureViewForEditingIfNeeded() {
+        // Check the helper to determine whether an editiing state should be assumed.
+        adjustViewForKeyboard(SigninEditingState.signinEditingStateActive)
+        if SigninEditingState.signinEditingStateActive {
+            emailTextField.becomeFirstResponder()
+        }
+    }
+
+
+    // MARK: - Instance Methods
+
+
+    /// Makes the call to retrieve Safari shared credentials if they exist.
+    ///
+    func fetchSharedWebCredentialsIfAvailable() {
+        didRequestSafariSharedCredentials = true
+        SigninHelpers.requestSharedWebCredentials { [weak self] (found, username, password) in
+            self?.handleFetchedWebCredentials(found, username: username, password: password)
+        }
+    }
+
+
+    /// Handles Safari shared credentials if any where found.
+    ///
+    /// - Parameters:
+    ///     - found: True if credentails were found.
+    ///     - username: The selected username or nil.
+    ///     - password: The selected password or nil.
+    ///
+    func handleFetchedWebCredentials(_ found: Bool, username: String?, password: String?) {
+        didFindSafariSharedCredentials = found
+
+        guard let username = username, let password = password else {
+            return
+        }
+
+        // Update the login fields
+        loginFields.username = username
+        loginFields.password = password
+
+        // Persist credentials as autofilled credentials so we can update them later if needed.
+        loginFields.safariStoredUsernameHash = username.hash
+        loginFields.safariStoredPasswordHash = password.hash
+
+        signinWithUsernamePassword(false)
+
+        WPAppAnalytics.track(.loginAutoFillCredentialsFilled)
+    }
+
+
+    /// Displays the wpcom sign in form, optionally telling it to immedately make
+    /// the call to authenticate with the available credentials.
+    ///
+    /// - Parameters:
+    ///     - immediateSignin: True if the newly loaded controller should immedately attempt
+    ///                        to authenticate the user with the available credentails.  Default is `false`.
+    ///
+    func signinWithUsernamePassword(_ immediateSignin: Bool = false) {
```

Comment:
This should be `loginWith…` now 😜. And what about naming that parameter… `loginToWPCom(immediately: Bool)`?